.gitlab/includes/common_spack_pipeline.yml

```diff
@@ -89,4 +105,3 @@ base_spack_image:
     - export CTEST_XML=$PWD/ctest.xml
     - trap "${SOURCE_DIR}/.gitlab/scripts/collect_ctest_metrics.sh ${CTEST_XML}" EXIT
     - spack -e pika_ci build-env $spack_spec -- bash -c "ctest --output-junit ${CTEST_XML} --label-exclude COMPILE_ONLY --test-dir ${BUILD_DIR} -j$(($(nproc)/2)) --timeout 120 --output-on-failure --no-compress-output --no-tests=error"
-  image: $PERSIST_IMAGE_NAME
```

Comment:
How does the test stage now know which image to use if `image` is not set explicitly?

Reply:
Ah, I see. I missed that, thanks!

All right, let's go with this for now and if we become wiser in the future we can change it.

My only very minor aesthetic nitpick would be to name the variables `PERSIST_IMAGE_NAME_{RELEASE,DEBUG}`, i.e. keep it all-caps if possible.